### ISSUE TITLE
updateNodeData does not run consistently in 12.0.0-next.12, next.13, next.14

### ISSUE THREAD

**Issue by NawarA** [xyflow/xyflow#4147]
### Describe the Bug

updateNodeData works consistently in `12.0.0-next.11`

Consistently means it always works as expected.

So 
```js
// updates the node's data property 100% of the time
updateNodeData( id, data )
``` 

Ever since `12.0.0-next.12` the command `updateNodeData( id, data )` no longer updates the node data property consistently.

So 
```js
// SOMETIMES update the node's data property
updateNodeData( id, data )
``` 

When I downgrade from 12.0.0-next.14 to 12.0.0-next.11, the update function works again.

### Steps to Reproduce the Bug or Issue

See code samples above...sorry I didn't create a separate repo with completely isolated code, but hopefully being to replicate the error is easy for you...just run the `updateNodeData( id, data )`.

By the way, in case it's helpful to mention, I usually update multiple nodeId's during the same execution, not sure if that's related to the root of the problem, but thought I'd mention it, as its part my setup of the issue.

### Expected behavior

```js
// updates the node's data property 100% of the time
updateNodeData( id, data )
``` 

### Platform

- OS: Windows 12
- Browser: Chrome
- Version: 123.0

**Comment by moklick** [comment]
Thanks for helping with the 12 release. We will try to find out what is wrong here.

**Comment by moklick** [comment]
I can't reproduce this. Can you create a codesandbox for this? Maybe you can use this as a starting point https://codesandbox.io/p/sandbox/reactflow-reactive-flow-34lk98?file=/src/Flow.tsx

**Comment by NawarA** [comment]
I setup a codesandbox env, but I'm unable to replicate the bug so far.

A few observations:
- When I downgrade from 12.0.0-14 to 12.0.0-11, the issue disappears
- When I keep 12.0.0-14, run multiple `updateNodeData` calls at the same exact same millisecond the bug occurs, but if I use a settimeout to space `updateNodeData` calls by about 100ms, the problem disappears. 
- I'm going to investigate further this weekend to try to trace the root, and make the bug replicable.

**Comment by AndreBurkett** [comment]
I believe that I am experiencing the same issue in next.14. For me, the issue occurs when connecting nodes in parallel. I've included a codesandbox below which demonstrates that only 1 path from the "Hello" textbox gets updated. A few things I've noticed, when connecting nodes in the following order A->B->C->D and A->E->F, G, with node A connected to both B and E:
- Updating node A will update all nodes along one path A->B->C->D. However, only Node E will be updated along the second path. F and G will not receive updates.
- Disconnecting and then reconnecting the A-E edge will properly update nodes F and G

[https://codesandbox.io/p/sandbox/reactflow-reactive-flow-forked-c87mwh](url)
